File: BE/health-service/src/main/java/com/s206/health/exercise/dto/response/MonthlyExerciseStatsResponse.java
package com.s206.health.exercise.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.YearMonth;
import java.util.List;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MonthlyExerciseStatsResponse {
    private List<MonthlyExercise> monthlyExercises;

    @Getter
    @Builder
    public static class MonthlyExercise {
        private YearMonth yearMonth;
        private Double avgDailyCalories;
        private Integer totalExerciseCount;
        private Integer totalCalories;
    }
}

File: FE/mobile/src/main/java/com/example/diaviseo/network/exercise/dto/res/MonthlyExerciseStatsResponse.kt
package com.example.diaviseo.network.exercise.dto.res

data class MonthlyExerciseStatsResponse(
    val monthlyExercises: List<MonthlyExercise>
) {
    data class MonthlyExercise(
        val yearMonth: String,
        val avgDailyCalories: Double,
        val totalExerciseCount: Int?,
        val totalCalories: Int?
    )
}
